src/auth-service/models/User.js

```
@@ -1159,48 +1159,119 @@
+        const isAdmin =
+          groupRole.userType === "user" ||
+          (groupRole.role && groupRole.role.role_name === "admin");
+
+        const hasPermission =
+          groupRole.role &&
+          groupRole.role.role_permissions &&
+          groupRole.role.role_permissions.some(
+            (p) => p === "view:private_cohorts" || p === "manage:cohorts"
+          );
+
+        if (isAdmin || hasPermission) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)